adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,960 @@
+                OptionHandler_Destroy(result);
+                result = NULL;
+            }
+            else
+            {
+                /*all is fine, all interesting options have been saved*/
+                /*return as is*/
+            }
+        }
+    }
+    return result;
+}
+
+static const IO_INTERFACE_DESCRIPTION tlsio_openssl_interface_description =
+{
+    tlsio_openssl_retrieveoptions,
+    tlsio_openssl_create,
+    tlsio_openssl_destroy,
+    tlsio_openssl_open,
+    tlsio_openssl_close,
+    tlsio_openssl_send,
+    tlsio_openssl_dowork,
+    tlsio_openssl_setoption
+};
+
+static LOCK_HANDLE * openssl_locks = NULL;
+
+
+static void openssl_lock_unlock_helper(LOCK_HANDLE lock, int lock_mode, const char* file, int line)
+{
+
+}
+
+static struct CRYPTO_dynlock_value* openssl_dynamic_locks_create_cb(const char* file, int line)
+{
+    struct CRYPTO_dynlock_value* result;
+    return result;
+}
+
+static void openssl_dynamic_locks_lock_unlock_cb(int lock_mode, struct CRYPTO_dynlock_value* dynlock_value, const char* file, int line)
+{
+    openssl_lock_unlock_helper(dynlock_value->lock, lock_mode, file, line);
+}
+
+static void openssl_dynamic_locks_destroy_cb(struct CRYPTO_dynlock_value* dynlock_value, const char* file, int line)
+{
+    Lock_Deinit(dynlock_value->lock);
+    free(dynlock_value);
+}
+
+static void openssl_dynamic_locks_uninstall(void)
+{
+}
+
+static void openssl_dynamic_locks_install(void)
+{
+}
+
+static void openssl_static_locks_lock_unlock_cb(int lock_mode, int lock_index, const char * file, int line)
+{
+}
+
+static void openssl_static_locks_uninstall(void)
+{
+    if (openssl_locks != NULL)
+    {
+        CRYPTO_set_locking_callback(NULL);
+        int i = 0;
+        for(i = 0; i < CRYPTO_num_locks(); i++)
+        {
+            if (openssl_locks[i] != NULL)
+            {
+                Lock_Deinit(openssl_locks[i]);
+            }
+        }
+        
+        free(openssl_locks);
+        openssl_locks = NULL;
+    }
+    else
+    {
+        LogError("Locks already uninstalled");
+    }
+}
+
+static void indicate_error(TLS_IO_INSTANCE* tls_io_instance)
+{
+    if (tls_io_instance->on_io_error == NULL)
+    {
+        LogError("NULL on_io_error.");
+    }
+    else
+    {
+        tls_io_instance->on_io_error(tls_io_instance->on_io_error_context);
+    }
+}
+
+static void indicate_open_complete(TLS_IO_INSTANCE* tls_io_instance, IO_OPEN_RESULT open_result)
+{
+    if (tls_io_instance->on_io_open_complete == NULL)
+    {
+        LogError("NULL on_io_open_complete.");
+    }
+    else
+    {
+        tls_io_instance->on_io_open_complete(tls_io_instance->on_io_open_complete_context, open_result);
+    }
+}
+
+
+static int httpconn_net_errno(int fd)
+{
+    int sock_errno = 0;
+    u32_t optlen = sizeof(sock_errno);
+    getsockopt(fd, SOL_SOCKET, SO_ERROR, &sock_errno, &optlen);
+    return sock_errno;
+}
+
+static void httpconn_set_non_block(int fd) 
+{
+  int flags = -1;
+  int error = 0;
```

Comment:
Can you please explain why initialization is not required here? I thought it's always good to initialize any variable defined.